### PR TITLE
Fix account switching

### LINKED ISSUE #218: Switching accounts intermittently broken
Switching accounts often results in empty sidebars and a 'No messages on server' message on the home screen. Seems to occur more often than not 

Configuration: Release build, on device

### PR DESCRIPTION
**Pull request by borisyankov**
Fixes #218 

Clears current state and requests new data from server.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@borisyankov, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/account/AccountPickScreen.js [hunk 1]
```diff
@@ -19,6 +20,14 @@ class AccountPickScreen extends React.Component {
     const { apiKey } = accounts[index];
     if (apiKey) {
       switchAccount(index); // Reset stream, message, user list
+
+      setTimeout(() => {
```
Comment (neerajwahi) [inline src/account/AccountPickScreen.js]: Why do we need the timeout and could we use `requestAnimationFrame` instead?

File: src/account/AccountPickScreen.js [hunk 2]
```diff
@@ -19,6 +20,14 @@ class AccountPickScreen extends React.Component {
...
+        const { auth, fetchEvents, fetchUsersAndStatus, fetchMessages } = this.props;
+
+        fetchEvents(auth);
```
Comment (neerajwahi) [inline src/account/AccountPickScreen.js]: These 3 lines are repeated in `MainScreen.js` so maybe we should refactor them out to avoid duplication

**Comment by borisyankov** [comment]
Code to reset the store existed, but the main component did not re render so it did not request new data.
I decided this is a good enough solution, though a refactoring will improve it.
We should probably invalidate the MainScreen component or somehow extract the code in another function later.

File: src/account/AccountPickScreen.js (hunk 1 above)
Comment (neerajwahi) [inline src/account/AccountPickScreen.js]: Ping on this question

Comment (borisyankov) [inline src/account/AccountPickScreen.js]: Probably not good reason. Will try with `requestAnimationFrame`

**Comment by borisyankov** [comment]
Fixed.

File: src/main/requestInitialServerData.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export default ({
+  auth,
+  fetchEvents,
+  fetchUsersAndStatus,
+  backgroundFetchMessages,
+  fetchMessages,
+}) => {
+  fetchEvents(auth);
+  fetchUsersAndStatus(auth);
+  fetchMessages(auth, Number.MAX_SAFE_INTEGER, 20, 0, homeNarrow(), [true, true]);
```
Comment (neerajwahi) [inline src/main/requestInitialServerData.js]: Looks like there was a merge conflict here. `[true, true]` should be `{ older: true, newer: true }`

**Comment by neerajwahi** [comment]
I fixed the comment above and merged this PR.

**Pull request closed without merging**